Library/Homebrew/tap.rb

```diff
@@ -570,7 +570,7 @@ def formula_file?(file)
     file = file.expand_path(path)
     return false unless ruby_file?(file)
 
-    file.to_s.start_with?("#{formula_dir}/")
+    file.to_s.start_with?("#{formula_dir}/") && !file.to_s.start_with?("#{cask_dir}/")
```

Comment:
```suggestion
    return false if cask_file?(file)

    file.to_s.start_with?("#{formula_dir}/")
```

to avoid duplicating logic